notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "\n",
+      "    At most one of the input config or the update config may be left unspecified, in which case the\n",
+      "    other will be read from ``stdin``. If neither filename or format is specified for the update config, no\n",
+      "    update will be performed.\n",
+      "\n",
+      "    The output destination may be specified as a filesystem path. When it is not, it will be written to\n",
+      "    ``stdout``.\n",
+      "\n",
+      "    If ``values_needed`` is ``True``, a report of values needed to realize the config is logged. In\n",
+      "    ``dry_run`` mode, output is written to ``stderr``.\n",
+      "\n",
+      "    If ``total`` is ``True``, an exception will be raised if any Jinja2 variables/expressions cannot be\n",
+      "    rendered. Otherwise, such variables/expressions will be passed through unchanged in the output.\n",
+      "\n",
+      "    Recognized file extensions are: ini, nml, sh, yaml\n",
+      "\n",
+      "    :param input_config: Input config file (``None`` => read ``stdin``).\n",
+      "    :param input_format: Format of the input config (optional if file's extension is recognized).\n",
+      "    :param update_config: Update config file (``None`` => read ``stdin``).\n",
+      "    :param update_format: Format of the update config (optional if file's extension is recognized).\n",
+      "    :param output_file: Output config file (``None`` => write to ``stdout``).\n",
+      "    :param output_format: Format of the output config (optional if file's extension is recognized).\n",
+      "    :param key_path: Path through keys to the desired output block.\n",
+      "    :param values_needed: Report complete, missing, and template values.\n",
+      "    :param total: Require rendering of all Jinja2 variables/expressions.\n",
+      "    :param dry_run: Log output instead of writing to output.\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: The ``dict`` representation of the realized config.\n",
+      "    :raises: UWConfigRealizeError if ``total`` is ``True`` and any Jinja2 variable/expression was not rendered.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.realize)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "501da514-a654-4511-928f-b2ad7db102b2",
+   "metadata": {},
+   "source": [
+    "The `input_config` parameter takes a config from a string path, <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object, Python `dict`, or `Config` objects like the `YAMLConfig` object from the <a href=\"#Getting-Config-Objects\">Getting Config Objects</a> section. The format of the input config can be specified with `input_format`. Configs are written to `stdout` if `output_file` is set to `None` or to a file specified by `output_file` and `output_format`. Providing the formats of input and output config files is optional if their file extensions are ini, nml, sh, or yaml.\n",
```

Comment:
```suggestion
    "The `input_config` parameter takes a config from a string path, <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object, Python `dict`, or UW `Config` objects like the `YAMLConfig` object from the <a href=\"#Getting-Config-Objects\">Getting Config Objects</a> section. The format of the input config can be specified with `input_format`. Configs are written to `stdout` if `output_file` is set to `None` or to a file specified by `output_file` and `output_format`. Providing the formats of input and output config files is optional if their file extensions are `ini`, `nml`, `sh`, or `yaml`.\n",
```

Just a proposed addition of `UW`. Maybe it doesn't make sense given you're immediately explaining what kind of objects.

It might be nice to use the code highlight on the file formats at the end of the text, though.

Reply:
Sounds good to me, I added UW before `Config` here and in other places I used this.